contributors/design-proposals/mount-options.md

```diff
@@ -0,0 +1,76 @@
+# Mount options for mountable volume types
+
+## Goal
+
+Enable Kubernetes admins to specify mount options with mountable volumes
+such as  - `nfs`, `glusterfs` or `aws-ebs` etc.
+
+## Motivation
+
+We currently support network filesystems: NFS, Glusterfs, Ceph FS, SMB (Azure file), Quobytes, and local filesystems such as ext[3|4] and XFS.
+
+Mount time options that are operationally important and have no security implications should be suppported. Examples are NFS's TCP mode, versions, lock mode, caching mode; Glusterfs's caching mode; SMB's version, locking, id mapping; and more.
+
+## Design
+
+Mount options can be specified as `mountOption`  as part of PV. As an example:
+
+``` yaml
+ apiVersion: v1
+  kind: PersistentVolume
+  metadata:
+    name: pv0003
+  spec:
+    capacity:
+      storage: 5Gi
+    accessModes:
+      - ReadWriteOnce
+    persistentVolumeReclaimPolicy: Recycle
+    mountOption: ["hard","nolock","nfsvers=3"]
+    nfs:
+      path: /tmp
+      server: 172.17.0.2
+```
+
+Or it can be specified as part of StorageClass definition:
+
+```yaml
+apiVersion: storage.k8s.io/v1beta1
+kind: StorageClass
+metadata:
+  name: slow
+provisioner: kubernetes.io/glusterfs
+parameters:
+  resturl: "http://127.0.0.1:8081"
+  restauthenabled: "true"
+  restuser: "admin"
+  secretNamespace: "default"
+  secretName: "heketi-secret"
+  mountOption: ["volume-name=foobar"]
```

Comment:
Good catch about `map[string]string` vs `map[string][]string`.   I am thinking this will become a non-issue if we move this field to its own place. 

There are other PRs such as - https://github.com/kubernetes/kubernetes/pull/40805 which do something similar.

I am as well somewhat hesitant to piggyback on `parameter` field since it hinders discoverability of API (just a bag of key/value pairs). cc @childsb